MovingBlocks/Terasology
# build: move `buildSrc` to `build-logic`

As a developer, when I add gestalt to my `libs/` workspace, IntelliJ IDEA collapses with a message about being unable to resolve `gestalt-module`.

I filed this as [IDEA-261518](https://youtrack.jetbrains.com/issue/IDEA-261518) and the recommendation from the gradle folks is to replace `buildSrc` with an included build.

Understanding why involves some gradle history.

Gradle made `buildSrc` a place to put code that's not on the same level as your (multi-) project build. It's the code that needs to build _before_ it figures out how to build the rest of your project.

Then came along composite builds, which let you include independent builds and use their results in your root project.

That has a lot of overlap with what `buildSrc` is for!

As composite builds become more useful, `buildSrc` becomes more of a relic, as it's less widely applicable. So they're having some trouble keeping feature parity between buildSrc and included builds. The plan is _eventually_ to make `buildSrc` just another included build, but it sounds like they expect that to take a while.

The current recommendation is to use an included build, even though it's not quite a drop-in replacement. That's why this PR has some more stuff creeping back in to the root project's `buildscript`: buildSrc used to add those to the environment by default.

For further reading, see https://github.com/gradle/gradle/issues/14713#issuecomment-708456079




### How to test

build stuff.

check out stuff to `libs/`, as described in [wiki:Using Locally Developed Libraries](https://github.com/MovingBlocks/Terasology/wiki/Using-Locally-Developed-Libraries).
